packages/agreement-email-sender/test/agreementEmailSender.integration.test.ts

```diff
@@ -0,0 +1,283 @@
+import { describe, expect, it, vi } from "vitest";
+import axios from "axios";
+import {
+  AgreementId,
+  DescriptorId,
+  EServiceId,
+  TenantId,
+  UserId,
+  generateId,
+  genericInternalError,
+  toAgreementV2,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import {
+  getMockAgreement,
+  getMockDescriptor,
+  getMockEService,
+  getMockTenant,
+} from "pagopa-interop-commons-test";
+import { InstitutionResponse } from "pagopa-interop-selfcare-v2-client";
+import { sendAgreementEmail } from "../src/services/agreementEmailSenderService.js";
+import {
+  addOneAgreement,
+  addOneEService,
+  addOneTenant,
+  emailManager,
+  emailManagerConfig,
+  readModelService,
+  selfcareV2ClientMock,
+} from "./utils.js";
+
+describe("agreement email sender", () => {
+  it("should send an email on AgreementSubmitted", async () => {
+    const tenant = { ...getMockTenant(), name: "Tenant" };
+    await addOneTenant(tenant);
+    const agreementId = unsafeBrandId<DescriptorId>(
+      "a8f059b2-3931-4802-ad9d-203936e98c22"
+    );
+    const descriptror = { ...getMockDescriptor(), version: "1" };
+    const eservice = {
+      ...getMockEService(),
+      name: "eService",
+      descriptors: [descriptror],
+    };
+    await addOneEService(eservice);
+    const agreement = {
+      ...getMockAgreement(eservice.id, tenant.id),
+      id: unsafeBrandId<AgreementId>(agreementId),
+      stamps: { activation: { when: new Date(0), who: generateId<UserId>() } },
+      producerId: tenant.id,
+      descriptorId: descriptror.id,
+    };
```

Comment:
See the comment above - no need to generate the agreement ID before, `getMockAgreement` already does that

```suggestion
    const agreement = {
      ...getMockAgreement(eservice.id, tenant.id),
      stamps: { activation: { when: new Date(0), who: generateId<UserId>() } },
      producerId: tenant.id,
      descriptorId: descriptror.id,
    };
```

Reply:
Same for other tests

Reply:
Didn't notice that the agreement id needs to be hardcoded because it's present in the email, ignore the comment